lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -236,8 +238,21 @@ func (p DockerEventPublisher) PublishEvent(event plevents.BaseEvent) {
 			attrs["exitCode"] = code
 			log.Infof("Sending die event for container %s - code: %s", vc.ContainerID, code)
 			actor := CreateContainerEventActorWithAttributes(vc, attrs)
-
 			EventService().Log(containerDieEvent, eventtypes.ContainerEventType, actor)
+
+			// auto-remove if required
+			if vc.HostConfig.AutoRemove {
+				config := &types.ContainerRmConfig{
+					ForceRemove:  true,
+					RemoveVolume: true,
```

Comment:
@hickeng As for nfs we will need more changes for that I fear. We need a consistent "attach" history so that we know who is using a volume.